pennylane/templates/layers/basic_entangler.py

```diff
@@ -0,0 +1,161 @@
+
+        The template is used inside a qnode:
+
+        .. code-block:: python
+
+            import pennylane as qml
+            from pennylane.templates import BasicEntanglerLayers
+            from math import pi
+
+            n_wires = 3
+            dev = qml.device('default.qubit', wires=n_wires)
+
+            @qml.qnode(dev)
+            def circuit(weights):
+                BasicEntanglerLayers(weights=weights, wires=range(n_wires))
+                return [qml.expval(qml.PauliZ(wires=i)) for i in range(n_wires)]
+
+        >>> circuit([[pi, pi, pi]])
+        [1., 1., -1.]
+
+        **Parameter initialization function**
+
+        The :mod:`~pennylane.init` module has two parameter initialization functions, ``basic_entangler_layers_normal``
+        and ``basic_entangler_layers_uniform``.
+
+        .. code-block:: python
+
+            from pennylane.init import basic_entangler_layers_normal
+
+            n_layers = 4
+            weights = basic_entangler_layers_normal(n_layers=n_layers, n_wires=n_wires)
+
+            circuit(weights)
+
+
+        **No periodic boundary for two wires**
+
+        When using two wires, the convention is to drop the periodic boundary condition.
+        This means that the connection from the second to the first wire is omitted.
+
+        .. code-block:: python
+
+            n_wires = 2
+            dev = qml.device('default.qubit', wires=n_wires)
+
+            @qml.qnode(dev)
+            def circuit(weights):
+                BasicEntanglerLayers(weights=weights, wires=range(n_wires))
+                return [qml.expval(qml.PauliZ(wires=i)) for i in range(n_wires)]
+
+        >>> circuit([[pi, pi]])
+        [-1, 1]
+
+
+        **Changing the rotation gate**
+
+        Any single-qubit gate can be used as a rotation gate, as long as it only takes a single parameter. The default is the ``RX`` gate.
+
+        .. code-block:: python
+
+            @qml.qnode(dev)
+            def circuit(weights):
+                BasicEntanglerLayers(weights=weights, wires=range(n_wires), rotation=qml.RZ)
+                return [qml.expval(qml.PauliZ(wires=i)) for i in range(n_wires)]
+
+        Accidentally using a gate that expects more parameters throws a
+        ``ValueError: Wrong number of parameters``.
+    """
+
+    #############
+    # Input checks
+
+    if rotation is None:
+        rotation = RX
+
+    _check_no_variable(rotation, msg="'rotation' cannot be differentiable")
+
+    wires = _check_wires(wires)
+
+    repeat = _check_number_of_layers([weights])
+
+    expected_shape = (repeat, len(wires))
+    _check_shape(
+        weights,
+        expected_shape,
+        msg="'weights' must be of shape {}; got {}" "".format(expected_shape, _get_shape(weights)),
+    )
+
+    ###############
+
+    for layer in range(repeat):
+
+        broadcast(unitary=rotation, pattern="single", wires=wires, parameters=weights[layer])
+        broadcast(unitary=CNOT, pattern="ring", wires=wires)
```

Comment:
Still being fairly new to PennyLane, I find this quite awesome. A very compact, nice and clear way of writing these circuits. 😄 👍